.github/workflows/publish.yml

```diff
@@ -0,0 +1,37 @@
+name: Publish
+on:
+  workflow_dispatch:
+permissions:
+  contents: write
+jobs:
+  publish:
+    runs-on: windows-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@main
+      with:
+        fetch-depth: 0
+    - name: Setup .NET Core SDK
+      uses: actions/setup-dotnet@main
+      with:
+        dotnet-version: '7.x.x'
+    - name: Install GitVersion
+      uses: gittools/actions/gitversion/setup@main
+      with:
+        versionSpec: '5.x'
+    - name: Determine Version
+      uses: gittools/actions/gitversion/execute@main
+    - name: Publish
+      run: dotnet publish DTALauncherStub.csproj --configuration Release --no-self-contained -p:PublishSingleFile=true -r win-x86 -p:DebugType=embedded -p:AssemblyVersion=$env:GitVersion_AssemblySemVer -p:FileVersion=$env:GitVersion_AssemblySemFileVer -p:InformationalVersion=$env:GitVersion_InformationalVersion
+    - name: Zip
+      run: 7z a -r ${{ format('DTALauncherStub-v{0}-net7.0-windows-x86.zip', env.GitVersion_SemVer) }} ./bin/Release/net7.0-windows/win-x86/publish/*.*
+    - name: Prerelease
+      if: ${{ env.GitVersion_PreReleaseTag != '' }}
+      run: gh release create ${{ format('v{0}', env.GitVersion_SemVer) }} (get-item *.zip) --generate-notes --target ${{ github.sha }} --prerelease
+      env:
+        GH_TOKEN: ${{ github.token }}
+    - name: Release
+      if: ${{ env.GitVersion_PreReleaseTag == '' }}
+      run: gh release create ${{ format('v{0}', env.GitVersion_SemVer) }} (get-item *.zip) --generate-notes --target ${{ github.sha }}
+      env:
+        GH_TOKEN: ${{ github.token }}
```

Comment:
Style nitpick - I think it's better to actually separate the steps with a newlines for each new step, the hyphen is easy to miss IMO.